pyproject.toml

```
@@ -35,7 +35,7 @@ cachecontrol = { version = "^0.12.4", extras = ["filecache"] }
 pkginfo = "^1.4"
 html5lib = "^1.0"
 shellingham = "^1.1"
-tomlkit = "^0.5.11"
+tomlkit = ">=0.5.0,<1.0.0"
```

```suggestion
tomlkit = "^0.7.0"
```

I want to stay on the conservative side of things but also ensure we have the latest version available at the time which is the 0.7.0 release, published today: https://github.com/sdispater/tomlkit/pull/104

---

@sdispater with dependencies like `tomlkit` we might end up having issues similar to #2752 where downstream poetry will be harder to package. If we do not rely on specific features or  are not affected by any bugs in question, we should consider leaving it at the lowest compatible version.

---

@abn Unfortunately, we don't have a lot of choices if we want to support TOML `1.0.0rc1` fully. The `0.6.0` release introduced the heterogeneous arrays support and the `0.7.0` release fixed a lot of issues with `1.0.0rc1` support which were not necessarily backwards compatible if people were relying on the previous, buggy behavior.

---

I reckon that is solid rationale to require 0.7.0.

---

@sdispater are you okay with `>=0.7.0,<1.0.0`? I think `>=0.7.0,<0.8.0` might be a too restrictive.